### PR TITLE
🌙 Add GCP cloud functions MQL resource

### PR DESCRIPTION
**Pull request by imilchev**
GCP Cloud Function resource:
- `gcp.project.cloudFunctions{ * }`

### CODE REVIEW

File: resources/packs/gcp/gcp.lr [hunk 1]
```diff
@@ -1358,3 +1360,73 @@ private gcp.project.loggingservice.bucket.indexConfig {
...
+  name string
+  // Cloud function description
+  description string
+  // Location of the archive with the function's source code
+  sourceArchiveUrl string
+  // Repository reference for the function's source code
+  sourceRepository dict
+  // Location of the upload with the function's source code
+  sourceUploadUrl string
+  // HTTPS endpoint of source that can triggered via URL
```
Comment (misterpantz) [inline resources/packs/gcp/gcp.lr]:
```suggestion
  // HTTPS endpoint of source that can be triggered via URL
```

File: resources/packs/gcp/gcp.lr [hunk 2]
```diff
@@ -1358,3 +1360,73 @@ private gcp.project.loggingservice.bucket.indexConfig {
...
+  availableMemoryMb int
+  // Email of the function's service account
+  serviceAccountEmail string
+  // Update timestamp
+  updated time
+  // Version identifier of the cloud function
+  versionId int
+  // Labels associated with this cloud function
+  labels map[string]string
+  // Environment variables that shall be available during function execution
```
Comment (misterpantz) [inline resources/packs/gcp/gcp.lr]:
```suggestion
  // Environment variables that are available during function execution
```

Although I LOVE that you wrote "shall"!!! <3

File: resources/packs/gcp/gcp.lr [hunk 3]
```diff
@@ -1358,3 +1360,73 @@ private gcp.project.loggingservice.bucket.indexConfig {
...
+  sourceRepository dict
+  // Location of the upload with the function's source code
+  sourceUploadUrl string
+  // HTTPS endpoint of source that can triggered via URL
+  httpsTrigger dict
+  // Source that fires events in response to a condition in another service
+  eventTrigger dict
+  // Status of the function deployment
+  status string
+  // Name of the function (as defined in source code) that will be executed
```
Comment (misterpantz) [inline resources/packs/gcp/gcp.lr]:
```suggestion
  // Name of the function (as defined in source code) that is executed
```

Whenever possible, I try to use present tense. But please let me know if it really breaks the meaning!

File: resources/packs/gcp/gcp.lr [hunk 4]
```diff
@@ -1358,3 +1360,73 @@ private gcp.project.loggingservice.bucket.indexConfig {
...
+  serviceAccountEmail string
+  // Update timestamp
+  updated time
+  // Version identifier of the cloud function
+  versionId int
+  // Labels associated with this cloud function
+  labels map[string]string
+  // Environment variables that shall be available during function execution
+  envVars map[string]string
+  // Build environment variables that shall be available during build time
```
Comment (misterpantz) [inline resources/packs/gcp/gcp.lr]:
```suggestion
  // Build environment variables that are available during build time
```

File: resources/packs/gcp/gcp.lr [hunk 5]
```diff
@@ -1358,3 +1360,73 @@ private gcp.project.loggingservice.bucket.indexConfig {
...
+  buildEnvVars map[string]string
+  // VPC network that this cloud function can connect to
+  network string
+  // Limit on the maximum number of function instances that may coexist at a given time
```
Comment (misterpantz) [inline resources/packs/gcp/gcp.lr]:
```suggestion
  // Maximum number of function instances that may coexist at a given time
```

File: resources/packs/gcp/gcp.lr [hunk 6]
```diff
@@ -1358,3 +1360,73 @@ private gcp.project.loggingservice.bucket.indexConfig {
...
+  maxInstances int
+  // Lower bound for the number function instances that may coexist at a given time
```
Comment (misterpantz) [inline resources/packs/gcp/gcp.lr]:
```suggestion
  // Lower bound for the number of function instances that may coexist at a given time
```

File: resources/packs/gcp/gcp.lr [hunk 7]
```diff
@@ -1358,3 +1360,73 @@ private gcp.project.loggingservice.bucket.indexConfig {
...
+  minInstances int
+  // VPC network connector that this cloud function can connect to
+  vpcConnector string
+  // Egress settings for the connector controlling what traffic is diverted
+  egressSettings string
+  // Ingress settings for the function controlling what traffic can reach
+  ingressSettings string
+  // Resource name of a KMS crypto key used to encrypt/decrypt function resources
+  kmsKeyName string
+  // Name of the cloud build custom worker pool that should be used to build the function
```
Comment (misterpantz) [inline resources/packs/gcp/gcp.lr]:
```suggestion
  // Name of the Cloud Build custom WorkerPool that should be used to build the function
```

File: resources/packs/gcp/gcp.lr [hunk 8]
```diff
@@ -1358,3 +1360,73 @@ private gcp.project.loggingservice.bucket.indexConfig {
...
+  buildWorkerPool string
+  // Cloud build ID of the latest successful deployment of the function
```
Comment (misterpantz) [inline resources/packs/gcp/gcp.lr]:
```suggestion
  // Cloud Build ID of the latest successful deployment of the function
```

File: resources/packs/gcp/gcp.lr [hunk 9]
```diff
@@ -1358,3 +1360,73 @@ private gcp.project.loggingservice.bucket.indexConfig {
...
+  buildId string
+  // Cloud build name of the function deployment
```
Comment (misterpantz) [inline resources/packs/gcp/gcp.lr]:
```suggestion
  // Cloud Build name of the function deployment
```

**Review by misterpantz — COMMENTED** [review]
Beauty!! 
Just a few suggestions. Thanks!

**Review by preslavgerchev — APPROVED** [review]

**Pull request merged**